src/video/wayland/SDL_waylandwindow.c

```diff
@@ -2139,7 +2139,9 @@ void Wayland_SetWindowPosition(_THIS, SDL_Window *window)
         xdg_popup_reposition(wind->shell_surface.xdg.roleobj.popup.popup,
                              wind->shell_surface.xdg.roleobj.popup.positioner,
                              0);
+        return 0;
     }
+    return SDL_Unsupported();
```

Comment:
I did a `SDL_Unsupported()` here, but perhaps it's better to do `SDL_SetError("wayland cannot position non-pop-up windows")`?
This might prevent people to look into this issue.

Reply:
Yeah, a more descriptive error is probably appropriate.

Reply:
I changed it to `wayland cannot position non-popup windows`.